### DISCUSSION TITLE
CRC check error

### DISCUSSION THREAD

**Discussion by jjirasek** [eModbus/eModbus#298] [Q&A]
Hello,
I want to control solar Inverter. Reading registers works ok, but I can't get writing to work.
Here is document https://github.com/lambda-ger/sofar-hyd-ep-remote-control (Remote Control Description for SOFARSOLAR Hybird Inverter HYD-1.pdf) where is control described.

For example I try (page 6) Force discharge 1000W:
With ModbusMessage msg I prepared message, then I added CRC ( RTUutils::addCRC(msg) ).
printMsg(msg) shows this:
01 10 11 87 00 06 0C 00 00 00 00 FF FF FC 18 FF FF FC 18 40 0B 
which is exactly the same as in the pdf.

but I always get Error 00001023: Message=CRC check error

Can you please advise me where can be problem ?

Thank you

this is part of code:
ModbusMessage msg(SOFAR_SLAVE_ID, WRITE_MULT_REGISTERS, sofarPassiveManualGdes, 6, 12, sofarDischarge1000);
RTUutils::addCRC(msg);
printMsg(msg);
err = RS485.addRequest(msg, (uint32_t)sofarPassiveManualGdes);

**Comment by Miq1** [comment]
Just omit the CRC addition - the eModbus library does take care of that.

**Reply by jjirasek** [reply]
I did it before, originaly I used addRequest. Because I had CRC errors, I tried what I said in first post. This also showed me that sequence of bytes is correct.

**Reply by Miq1** [reply]
I would suggest to leave the CRC to the lib and try to find the root cause for the CRC errors. If you do a plain ``addRequest`` with your address and values, what does the (ideally VERBOSE) log tell?
